package/base-files/files/sbin/sysupgrade

```diff
@@ -118,25 +118,40 @@ list_conffiles() {
 		BEGIN { conffiles = 0 }
 		/^Conffiles:/ { conffiles = 1; next }
 		!/^ / { conffiles = 0; next }
-		conffiles == 1 { print }
+		# in format: <space><filename><space><hash>
+		conffiles == 1 { print $(NF) "  " substr($0,2,length($0)-length($(NF))-2) }
 	' /usr/lib/opkg/status
 }
 
 list_changed_conffiles() {
-	# Cannot handle spaces in filenames - but opkg cannot either...
-	list_conffiles | while read file csum; do
+	list_conffiles | while read -r csum_file; do
+		file="$(echo "$csum_file" | cut -d" " -f3-)"
```

Comment:
Why is this code change needed? Why that inefficient `cut` usage? What is wrong with `read file csum` ?

Reply:
> Why is this code change needed?

Current code does not support space in filename. I'm testing with an extreme case but any space at all would break sysupgrade like "/etc/my system/conf":

```
mkdir '/etc/A\ "!$HOME'"'"'  a*s?d[123] /'
touch '/etc/A\ "!$HOME'"'"'  a*s?d[123] /test_file'
echo '/etc/A\ "!$HOME'"'"'  a*s?d[123] /test_file' >>/etc/sysupgrade.conf
sysupgrade -l | grep test_file
```

I swapped file csum order to guarantee that the first matched "field separator" is not part of path. I used double space as "field separator" to match what sha256sum expects, so I do not need to build it later.

> What is wrong with `read file csum` ?

At least:
1. files will have only the first part of the filepath until it match any IFS. csum will have the remaining line.
2. any valid IFS will be converted and merged in a single space (tab, multiple spaces)
3. It is missing '-r', so \ is consumed by read

> Why that inefficient `cut` usage?

You are right. I overlooked it here. I pushed a new version that uses variable expansion "${csum_file#*  }", which is 300x faster.